bokeh/document.py

```diff
@@ -241,6 +241,27 @@ def add_periodic_callback(self, callback, period_milliseconds):
                               period_milliseconds)
         return self._add_session_callback(cb, callback, one_shot=False)
 
+    def _get_namespace(self):
+        from .models import Namespace
+
+        for model in self._roots:
+            if isinstance(model, Namespace):
+                return model
+        else:
+            model = Namespace()
+            self._roots.append(model)
+            return model
+
+    def set_var(self, name, value):
+        '''
+        '''
+        self._get_namespace().vars[name] = value
+
+    def get_var(self, name):
+        '''
+        '''
+        return self._get_namespace().vars[name]
+
```

Comment:
A similar issue is coming up in some continued events work @jlstevens is working on for "document events" (like startup/busy/done/etc). Its the same situation where an object needs to be added as a root in order to interact with properties or events on it. I'd like to handle these cases consistently. What we discussed:
* always just add the object to `roots` at doc creation
* keep an internal reference (so no need to search for it in `_roots`
* disallow removing the object from roots (check in `remove`, throw an exception)

I'd like to do that here as well. Then also have a readonly `vars` property that defines `getattr` and `setattr` so that this works:

```
curdoc().vars.foo = 10
```